src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -695,7 +723,16 @@ class ErgoWalletActor(settings: ErgoSettings,
       val targetBalance = outputs.map(_.value).sum
       val targetAssets = TransactionBuilder.collectOutputTokens(outputs.filterNot(bx => assetIssueBox.contains(bx)))
 
-      val selectionOpt = boxSelector.select(inputBoxes, filter, targetBalance, targetAssets)
+          val (inputBoxes, filter) = if (userInputs.nonEmpty) {
```

Comment:
Improper merging, please see how inputBoxes defined now, so please revert to

val selectionOpt = boxSelector.select(inputBoxes, filter, targetBalance, targetAssets)

and then reuse getBoxesToSpend in inputBoxes construction procedure as 

(getBoxesToSpend, walletFilter)

instead of 

val boxesToSpend = (registry.walletUnspentBoxes() ++ offChainRegistry.offChainBoxes).distinct
(boxesToSpend.toIterator, walletFilter)